packages/flutter/lib/src/material/input.dart

```diff
@@ -85,6 +209,14 @@ class _InputState extends State<Input> {
     ThemeData themeData = Theme.of(context);
     BuildContext focusContext = focusKey.currentContext;
     bool focused = focusContext != null && Focus.at(focusContext, autofocus: config.autofocus);
+    _FormFieldState fieldState = _FormFieldState.at(context, config.formField, this);
+    InputValue value = fieldState?.value ?? config.value;
+    ValueChanged<InputValue> onChanged = fieldState?.onChanged ?? config.onChanged;
+    ValueChanged<InputValue> onSubmitted = fieldState?.onSubmitted ?? config.onSubmitted;
```

Comment:
I would have expected the `??` to be reversed here.  If someone explicitly gives a config.value, shouldn't that override the default one provided by the form?